File: app/src/main/java/com/gaohui/nestedrecyclerview/java/adapter/CategoryPagerAdapter.java
package com.gaohui.nestedrecyclerview.java.adapter;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;


import com.gaohui.nestedrecyclerview.java.view.CategoryView;

import java.util.ArrayList;

public class CategoryPagerAdapter extends PagerAdapter {

    private ArrayList<CategoryView> mViewList;
    private ArrayList<String> mTabList;

    private CategoryView mCurrentPrimaryItem = null;


    public CategoryPagerAdapter(ArrayList<CategoryView> viewList,ArrayList<String> tabList) {
        mViewList = viewList;
        mTabList = tabList;
    }

    @Override
    public int getCount() {
        return mViewList.size();
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        CategoryView categoryView = mViewList.get(position);
        if(container == categoryView.getParent()) {
            container.removeView(categoryView);
        }
        container.addView(categoryView);
        return categoryView;
    }

    @Override
    public void setPrimaryItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        CategoryView categoryView = (CategoryView) object;
        if(categoryView  != mCurrentPrimaryItem) {
            if(mCurrentPrimaryItem  != null) {
                mCurrentPrimaryItem.onUserVisibleChange(false);
            }
        }
        categoryView.onUserVisibleChange(true);
        mCurrentPrimaryItem = categoryView;
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object o) {
        return view == o;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mTabList.get(position);
    }
}

File: app/src/main/java/com/gaohui/nestedrecyclerview/kotlin/adapter/CategoryPagerAdapter.kt
package com.gaohui.nestedrecyclerview.kotlin.adapter

import android.support.v4.view.PagerAdapter
import android.view.View
import android.view.ViewGroup
import com.gaohui.nestedrecyclerview.kotlin.CategoryView
import com.gaohui.nestedrecyclerview.kotlin.tab.DynamicTabBean
import com.gaohui.nestedrecyclerview.kotlin.tab.DynamicTabLayout

class CategoryPagerAdapter(
    private val viewList: ArrayList<CategoryView>,
    private val tabTitleList: ArrayList<String>
) : PagerAdapter(), DynamicTabLayout.DynamicTabProvider {

    private var mCurrentPrimaryItem: CategoryView? = null

    override fun getCount(): Int {
        return viewList.size
    }

    override fun isViewFromObject(view: View, obj: Any): Boolean {
        return view === obj
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = viewList[position]
        if (container == view.parent) {
            container.removeView(view)
        }
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        //container.removeView((View) object);
    }

    override fun setPrimaryItem(container: ViewGroup, position: Int, obj: Any) {
        val item = obj as CategoryView
        if(item != mCurrentPrimaryItem) {
            mCurrentPrimaryItem?.onUserVisibleChange(false)
        }
        item.onUserVisibleChange(true)
        mCurrentPrimaryItem = item
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return tabTitleList[position]
    }

    override fun getPageTitleItem(position: Int): DynamicTabBean? {
        return DynamicTabBean("推荐", "精品推荐")
    }
}